modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -455,9 +461,12 @@ class CV_EXPORTS_W VideoCapture
     @param filename name of the opened video file (eg. video.avi) or image sequence (eg.
     img_%02d.jpg, which will read samples like img_00.jpg, img_01.jpg, img_02.jpg, ...)
 
+    @param apiPreference preferred Capture API to use. Can be used to enforce a specific reader
+    implementation if multiple are available: e.g. CV_CAP_FFMPEG or CV_CAP_IMAGES
+
     The methods first call VideoCapture::release to close the already opened file or camera.
      */
-    CV_WRAP virtual bool open(const String& filename);
+    CV_WRAP virtual bool open(const String& filename, int apiPreference = CAP_ANY);
```

Comment:
I am not quite sure what exactly you mean here, but `CAP_ANY == CV_CAP_ANY` in the basic function. This is in turn handled by [the default case](https://github.com/Itseez/opencv/pull/4004/files#diff-025219446785708058c4098c73ac3067R372).